prow/lib.sh

```diff
@@ -113,10 +121,12 @@ function setup_kind_cluster() {
   if ! (kind delete cluster --name=istio-testing) > /dev/null; then
     echo "No existing kind cluster with name istio-testing. Continue..."
   fi
+
+  trap cleanup_kind_cluster EXIT
```

Comment:
Let's get more eyes on this.  Might also be worth adding ERR and perhaps some signal numbers to this.

@PiotrSikora @jplevyak @silentdai

Reply:
I was considering this, but we already do this same handling on L71 for Boskos (which is much worse if we fail to cleanup) and this is what kubernetes is doing so I figured it was sufficient

Reply:
I'm going to approve this one since catching EXIT is good and we can always improve it later by adding additional cases as needed